### PR TITLE
scala-native

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -46,17 +48,28 @@ ThisBuild / githubWorkflowPublish := Seq(
...
 // Aliases
 
-addCommandAlias("validate", ";clean;validateJVM;validateJS")
+addCommandAlias("validate", ";clean;validateJVM;validateJS;validateNative")
 addCommandAlias("validateJVM", ";buildJVM;mimaJVM;testJVM")
 addCommandAlias("validateJS", ";buildJS;mimaJS;testJS")
+addCommandAlias("validateNative", ";buildNative;mimaNative;testNative")
 addCommandAlias("buildJVM", ";derivingJVM/compile;testJVM/compile;typeableJVM/compile")
 addCommandAlias("buildJS", ";derivingJS/compile;testJS/compile;typeableJS/compile")
+addCommandAlias("buildNative", ";derivingNative/compile;testNative/compile")
```
Comment (joroKr21) [inline build.sbt:66]: We're missing typeable - is that on purpose?

**Review by joroKr21 — CHANGES_REQUESTED** [review]
Thanks, that's awesome! Check my comments for feedback.

File: build.sbt (hunk 1 above)
Comment (xuwei-k) [reply build.sbt:65]: Could not `typeableNative/Test/compile` 😢 

```
modules/typeable/src/test/scala/shapeless3/typeable/typeable.scala:315:32 
[error] 315 |    val co = o.cast[Option[Int]]
[error]     |                                ^
[error]     |Exception occurred while executing macro expansion.
[error]     |scala.scalanative.runtime.UndefinedBehaviorError
[error]     |	at scala.scalanative.runtime.package$.throwUndefined(package.scala:80)
[error]     |	at scala.scalanative.runtime.package$.intrinsic(package.scala:26)
[error]     |	at scala.scalanative.runtime.Intrinsics$.classFieldRawPtr(Intrinsics.scala:145)
[error]     |	at shapeless3.typeable.ReflectionUtils.Mirror(reflectionutils.scala:23)
[error]     |	at shapeless3.typeable.TypeableMacros$.mkSumTypeable$1(typeable.scala:337)
[error]     |	at shapeless3.typeable.TypeableMacros$.impl(typeable.scala:405)
[error]     |
[error]     | This location contains code that was inlined from typeable.scala:315
[error]     | This location contains code that was inlined from typeable.scala:250
```

Comment (joroKr21) [reply build.sbt:65]: Does it mean macros don't work with Scala native? 🤔

Comment (joroKr21) [reply build.sbt:65]: Didn't find anything in the release notes about macros: https://scala-native.readthedocs.io/en/latest/changelog/0.4.3-RC1.html - should we report it?

**Review by joroKr21 — APPROVED** [review]

**Comment by joroKr21** [comment]
Let's book the progress and try to figure out what's wrong with Typeable later

File: build.sbt (hunk 1 above)
Comment (WojciechMazur) [reply build.sbt:65]: Hey, it looks like a bug. `classFieldRawPtr` is currently only used in rewrites of LazyVals internal handling to allow for them to be run in Scala Native (code produced by Scala 3 for lazy vals is using sun.misc.Unsafe which would not be supported in Scala Native). I'll try to take a look at this issue and fix it before RC2

Comment (WojciechMazur) [reply build.sbt:65]: In general Scala Native should work well with macros, they're evaluated long before SN is added to the compilation pipeline

Comment (WojciechMazur) [reply build.sbt:65]: The fix for the compilation of typeable macros would be included in RC2, after applying the fix locally I was able to compile and successfully run all the tests.

Comment (xuwei-k) [reply build.sbt:65]: - https://github.com/typelevel/shapeless-3/pull/62 🎉

**Pull request merged**